src/connectivity/mod.rs

```diff
@@ -244,7 +243,11 @@ pub fn number_connected_components(graph: &graph::PyGraph) -> usize {
 #[pyfunction]
 #[pyo3(text_signature = "(graph, /)")]
 pub fn connected_components(graph: &graph::PyGraph) -> Vec<HashSet<usize>> {
-    conn_components::connected_components(&graph.graph)
+    let return_value: Vec<HashSet<usize>> = connectivity::connected_components(&graph.graph)
+        .into_iter()
+        .map(|res_map| res_map.into_iter().map(|x| x.index()).collect())
+        .collect();
+    return_value
```

Comment:
It'll probably be optimized away by the compiler but we don't need the `return_value` variable:

```suggestion
    connectivity::connected_components(&graph.graph)
        .into_iter()
        .map(|res_map| res_map.into_iter().map(|x| x.index()).collect())
        .collect()
```

Reply:
Done in https://github.com/Qiskit/retworkx/pull/595/commits/d5e5def3556d2221478c59cd334accb372573739.